### PR TITLE
Improve get_manifest() method -- `develop`

### PR DESCRIPTION
**Pull request by sujaypatil96**
This PR adds the feature/enhancement as described in [issue #214](https://github.com/Sage-Bionetworks/HTAN-coordination/issues/214).

Test feature by running:
`python3 examples/manifest_generator.py` and supplying different parameters based on the different use-cases.

_Note: Remember to make sure Synapse credentials have been set in `.synapseConfig` file._

### CODE REVIEW

File: schematic/manifest/generator.py [hunk 1]
```diff
@@ -178,27 +175,8 @@ def callback(request_id, response, exception):
         batch.execute()
 
 
-    def _get_valid_values_from_jsonschema_property(self, prop:dict) -> List[str]:
```
Comment (milen-sage) [inline schematic/manifest/generator.py:181]: Note: this function needs to stay in develop; similarly there is logic below that needs to be preserved. 

@sujaypatil96 any chance you can apply your changes over the latest code in develop?

**Comment by sujaypatil96** [comment]
@milen-sage I've updated this PR with the latest code from `develop`. I think it's good for another test/review (and merge).

File: config.yml [hunk 2]
```diff
@@ -1,9 +1,14 @@
 ---
   synapse:
     master_fileview: 'syn20446927'
+    manifest_folder: 'manifests'
     manifest_filename: 'manifests/synapse_storage_manifest.csv'
```
Comment (milen-sage) [inline config.yml:5]: I notice manifest folder got added as a config param; and we have manifest filename which still contains the 'manifests' folder in the path. Is this intentional?

Comment (sujaypatil96) [reply config.yml:5]: `manifest_filename` is being used in `schematic.synapse.store` on [L#83](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/synapse/store.py#L83), to explicitly point to the file in the `manifests` folder. The `manifests` folder (new parameter added to the config) is to point to the folder where the generated manifest (after running `$ python examples/manifest_generator.py`) should sit.

So in summary, the two config parameters are unrelated and we want to keep both of them.
